api/v1alpha5/virtualmachine_hardware_types.go

```diff
@@ -134,6 +208,19 @@ type SCSIControllerSpec struct {
 	Type SCSIControllerType `json:"type,omitempty"`
 }
 
+// MaxSlots returns the maximum number of slots per SCSI controller.
+func (c SCSIControllerSpec) MaxSlots() int32 {
+	return c.Type.MaxSlots()
+}
+
+func NewEmptySCSIControllerSpec() SCSIControllerSpec {
+	return SCSIControllerSpec{BusNumber: -1}
+}
+
+func (c SCSIControllerSpec) IsValid() bool {
```

Comment:
A few things:

* Exported symbols should have documentation! I have given this feedback a number of times. Please stop making me remind you.
* I do not like a generic `IsValid` function in the API. That is not the purpose of the API. The other helpers were about constants. This is business logic. Move it elsewhere.